src/components/ProjectPage/ProjectPage.tsx

```diff
@@ -31,6 +30,88 @@ const ModalOnEvent = dynamic(() => import('../ModalOnEvent'));
 const FilterCreateForm = dynamic(() => import('../FilterCreateForm/FilterCreateForm'));
 const FilterDeleteForm = dynamic(() => import('../FilterDeleteForm/FilterDeleteForm'));
 
+const PageProjectListItem: FC<
+    Omit<ComponentProps<typeof ProjectListItemCollapsable>, 'children' | 'goals'> & {
+        queryState: QueryState;
+        onTagClick?: React.ComponentProps<typeof GoalListItem>['onTagClick'];
+        onClickProvider?: (g: NonNullable<GoalByIdReturnType>) => MouseEventHandler<HTMLAnchorElement>;
+        selectedResolver?: (id: string) => boolean;
+    }
+> = ({ queryState, project, onClickProvider, onTagClick, selectedResolver, deep = 0, ...props }) => {
+    const [fetchGoalsEnabled, setFetchGoalsEnabled] = useState(false);
+
+    const { data: projectDeepInfo } = trpc.project.getDeepInfo.useQuery(
+        {
+            id: project.id,
+            ...queryState,
+        },
+        {
+            enabled: fetchGoalsEnabled,
+            keepPreviousData: true,
+            staleTime: refreshInterval,
+        },
+    );
+
+    const [fetchChildEnabled, setFetchChildEnabled] = useState(false);
+
+    const ids = useMemo(() => project?.children.map(({ id }) => id) || [], [project]);
+    const { data: childrenProjects = [], status } = trpc.project.getByIds.useQuery(ids, {
+        enabled: fetchChildEnabled,
+    });
+
+    const goals = useMemo(
```

Comment:
Can u explain this logic? Why we need to do that if we fetch goals of selected project?

Reply:
https://github.com/taskany-inc/issues/issues/1136